packages/node/src/ethereum/api.ethereum.ts

```diff
@@ -137,6 +137,44 @@ export class EthereumApi implements ApiWrapper<EthereumBlockWrapper> {
     this.supportsFinalization = supportsFinalization;
     this.chainId = network.chainId;
     this.name = network.name;
+
+    if (this.client instanceof WebSocketProvider) {
+      return;
+    }
+
+    logger.debug(
+      `checking maximum allowed batch request size for ${this.endpoint}...`,
+    );
+
+    // Find and set the maximum batch size allowed by JsonRpcBatchProvider
+    let batchSize = 10; // default maximum value
+    let minBatchSize = 1;
+    let maxBatchSize = batchSize;
+    let found = false;
+
+    while (!found && minBatchSize <= maxBatchSize) {
+      try {
+        // Test the current batch size with a dummy request (getBlock)
+        const batchPromises = [];
+        for (let i = 0; i < batchSize; i++) {
+          batchPromises.push(this.client.getBlock('latest'));
+        }
+        await Promise.all(batchPromises);
+
+        minBatchSize = batchSize;
+        batchSize = Math.min(batchSize * 2, maxBatchSize);
+      } catch (error) {
+        maxBatchSize = batchSize - 1;
+        batchSize = Math.floor((minBatchSize + maxBatchSize) / 2);
+      }
+
+      if (minBatchSize === maxBatchSize) {
+        found = true;
+      }
+    }
+
+    logger.debug(`batch request size for ${this.endpoint}: ${minBatchSize}`);
+    (this.client as JsonRpcBatchProvider).setBatchSize(minBatchSize);
```

Comment:
This doesn't seem like the right place to do this. It should live in the batch provider.